app/views/teachers/TeacherStudentView.coffee

```diff
@@ -0,0 +1,392 @@
+    # console.log (moment(session.get('changed')).format("LLLL"))
+    # Rerun template/jade file to display new last played string
+    @render()
+
+  updateLevelProgressMap: ->
+    return unless @courses.loaded and @levels.loaded and @sessions?.loaded and @user?.loaded
+
+    # Map levels to sessions once, so we don't have to search entire session list multiple times below
+    @levelSessionMap = {}
+    # for session in @sessions.models when session.get('creator') is @studentID
+    for session in @sessions.models when session.get('creator') is @studentID
+      @levelSessionMap[session.get('level').original] = session
+
+
+    # Create mapping of level to student progress
+    @levelProgressMap = {}
+    for versionedCourse in @classroom.get('courses') ? []
+      for versionedLevel in versionedCourse.levels
+        session = @levelSessionMap[versionedLevel.original]
+        if session
+          if session.get('state')?.complete
+            @levelProgressMap[versionedLevel.original] = 'complete'
+          else
+            @levelProgressMap[versionedLevel.original] = 'started'
+        else
+          @levelProgressMap[versionedLevel.original] = 'not started'
+
+  updateLevelDataMap: ->
+    return unless @courses.loaded and @levels.loaded and @sessions?.loaded
+
+    @levelData = []
+    for versionedCourse in @classroom.get('courses') ? []
+      course = _.find @courses.models, (c) => c.id is versionedCourse._id
+      for versionedLevel in versionedCourse.levels
+        @playTime = 0 # this and @timesPlayed should probably only count when the levels are completed
+        @timesPlayed = 0
+        @studentTime = 0
+        @levelProgress = 'not started'
+        for session in @sessions.models
+          if session.get('level').original == versionedLevel.original
+            # if @levelProgressMap[versionedLevel.original] == 'complete' # ideally, don't log sessions that aren't completed in the class
+            @playTime += session.get('playtime') or 0
+            @timesPlayed += 1
+            if session.get('creator') is @studentID
+              @studentTime = session.get('playtime') # this can be null, apparently.
+              if @levelProgressMap[versionedLevel.original] == 'complete'
+                @levelProgress = 'complete'
+              else if @levelProgressMap[versionedLevel.original] == 'started'
+                @levelProgress = 'started'
+        classAvg = if @timesPlayed and @timesPlayed > 0 then Math.round(@playTime / @timesPlayed) else 0 # only when someone other than the user has played
+        console.log (@timesPlayed)
+        @levelData.push {
+          levelID: versionedLevel.original
+          levelIndex: @classroom.getLevelNumber(versionedLevel.original)
+          levelName: versionedLevel.name
+          courseName: course.get('name')
+          courseID: course.get('_id')
+          class: classAvg
+          student: if @studentTime then @studentTime else 0
```

Comment:
Also, using "class" as a variable name is discourages because it's also a keyword. It's only allowed here at all because it's an attribute on an object.